src/include/nfs_server/ui.rb

```diff
@@ -200,6 +200,9 @@ def GetAllowedHosts(exports, expath, hosts, opts, allowed, fromedit)
         end
 
         if ret == :optsent
+          # Update the opts value when changed, otherwise the value validated
+          # later could be the cached one.
```

Comment:
What??
How does this fail without this code?

Reply:
When you add a new share it is initialize with defaults.... opts = ...

Then, just add a + for example to it, when you press ok it tries to validate opts... which has not been updated with the widget value. It is updated later... before storing

Reply:
## With the initialization of opts from the widget value

![Peek 2019-10-25 13-33](https://user-images.githubusercontent.com/7056681/67571857-e2f77280-f72c-11e9-86dc-a82ee5900973.gif)

## Without the line

![Peek 2019-10-25 13-36](https://user-images.githubusercontent.com/7056681/67571861-e559cc80-f72c-11e9-824c-f1b07d839a37.gif)

Reply:
Well, then this should be `opts = UI.QueryWidget(Id(:optsent), :Value)`, not `opts = ChangeWidget(:optsent, opts)` which I still don't understand how it works.

Reply:
Ups :) that was my error, I tested the change manually in the virtual machine, but then I copied the wrong line from below here :+1: 

Fixed

Reply:
Verified with the rpm instead of manual edition